backend_jobs/ingestion_removal/pipeline_lib/removal_pipeline_interface.py

```diff
@@ -0,0 +1,125 @@
+        the ingested runs array, neither or both.
+
+        Args:
+            query_provider: All docs with the image_provider as provider.
+            query_pipeline_run: All docs with the pipeline_run as pipeline_run.
+            parent_image_ref: A reference to the image doc in the image collection.
+            image_provider: The image provider we remove the images by.
+            pipeline_run: The image pipeline_run we remove the images by.
+        """
+        if len(query_provider.get()) == 0 and len(query_pipeline_run.get()) != 0:
+            self._update_provider_array(parent_image_ref, image_provider)
+        if len(query_provider.get()) != 0 and len(query_pipeline_run.get()) == 0:
+            self._update_pipeline_array(parent_image_ref, pipeline_run)
+        if len(query_provider.get()) == 0 and len(query_pipeline_run.get()) == 0:
+            self._update_provider_array(parent_image_ref, image_provider)
+            self._update_pipeline_array(parent_image_ref, pipeline_run)
+
+    def _update_provider_array(self, parent_image_ref, image_provider):
+        """ This function is in charge of removing the given image_provider from the ingested providers
+        array.
+
+        Args:
+            parent_image_ref: A reference to the image doc in the image collection.
+            image_provider: The image provider we remove the images by.
+        """
+        image_doc_dict = parent_image_ref.get().to_dict()
+        # Updates the provider array in the image collection.
+        providers_array = image_doc_dict[database_schema.COLLECTION_IMAGES_FIELD_INGESTED_PROVIDERS]
+        if image_provider in providers_array:
+            providers_array.remove(image_provider)
+        parent_image_ref.update({
+            database_schema.COLLECTION_IMAGES_FIELD_INGESTED_PROVIDERS: providers_array
+        })
+
+    def _update_pipeline_array(self, parent_image_ref, pipeline_run):
+        """ This function is in charge of removing the given pipeline_run from the ingested pipeline runs
+        array.
+
+        Args:
+            parent_image_ref: A reference to the image doc in the image collection.
+            pipeline_run: The image pipeline_run we remove the images by.
+        """
+        image_doc_dict = parent_image_ref.get().to_dict()
+        # Updates the pipeline array in the image collection.
+        pipeline_runs_array = image_doc_dict[database_schema.COLLECTION_IMAGES_FIELD_INGESTED_RUNS]
+        if pipeline_run in pipeline_runs_array:
+            pipeline_runs_array.remove(pipeline_run)
+        parent_image_ref.update({
+            database_schema.COLLECTION_IMAGES_FIELD_INGESTED_RUNS: pipeline_runs_array
+        })
+
+    def remove_image_doc_if_necessary(self, parent_image_id):
+        ''' This function is in charge of removing the original image if the subcollection
+        pipeline_runs does not contain any documents
```

Comment:
use """ not '''
Also, why do both ingested pipeline runs and ingested providers need to be empty?
Is it not enough if there is no pipeline run left in the array?